presto-main/src/test/java/com/facebook/presto/sql/planner/optimizations/TestMergeWindows.java

```diff
@@ -464,19 +468,21 @@ public void testNotMergeDifferentOrdering()
                                         .addFunction(functionCall("sum", COMMON_FRAME, ImmutableList.of(QUANTITY_ALIAS))),
                                 window(windowMatcherBuilder -> windowMatcherBuilder
                                                 .specification(specificationA)
-                                                .addFunction(functionCall("sum", COMMON_FRAME, ImmutableList.of(EXTENDEDPRICE_ALIAS)))
-                                                .addFunction(functionCall("sum", COMMON_FRAME, ImmutableList.of(DISCOUNT_ALIAS))),
+                                                .addFunction(functionCall("sum", UNSPECIFIED_FRAME, ImmutableList.of(EXTENDEDPRICE_ALIAS)))
```

Comment:
I am still not sure what your objection is. So let me clarify more. As per my interpretation, the above variable is referring to the frame obtained by calling `getFrame()` on `WindowNode.Function.FunctionCall.Window`. And in case of named windows, this would return an empty frame, since `WindowNode.Function.FunctionCall.Window` is nothing but `WindowName` in case of named windows and doesn't have embedded frame information. Hence specifying an empty frame (which is what `UNSPECIFIED_FRAME` is) is correct, IMO.

So are you objecting to the fact that `WindowNode.Function.FunctionCall.Window.getFrame()` is empty? For the `FunctionCall` class, which is in parser module, I thought the frame information is exactly what literally in the SQL query (without going through the analyzer). So for a function such as `SUM(discount) over w1`, the frame information is empty, where as for a function such as  `SUM(discount) over `(w1 ROWS BETWEEN UNBOUNDED PRECEDING AND CURRENT ROW)`, it is not. Isn't it?